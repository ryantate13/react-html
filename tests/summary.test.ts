import {summary} from '../src';

describe('summary', () => {
    it('renders with children', () => {
        expect(summary({}, 'TEST')).toMatchSnapshot();
    });

    it('renders with empty attributes', () => {
        expect(summary({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(summary()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(summary({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(summary({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(summary({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(summary({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(summary({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(summary({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(summary({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(summary({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(summary({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(summary({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(summary({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(summary({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(summary({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(summary({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(summary({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden="true"', () => {
        expect(summary({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden="false"', () => {
        expect(summary({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(summary({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(summary({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(summary({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(summary({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(summary({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(summary({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(summary({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(summary({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(summary({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(summary({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(summary({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(summary({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(summary({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope="true"', () => {
        expect(summary({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope="false"', () => {
        expect(summary({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(summary({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(summary({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(summary({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(summary({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(summary({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(summary({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style="[object Object]"', () => {
        expect(summary({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex="1"', () => {
        expect(summary({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(summary({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(summary({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(summary({translate: "no"})).toMatchSnapshot();
    });
});