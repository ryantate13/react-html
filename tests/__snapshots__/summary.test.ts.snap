// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`summary renders arrays of children 1`] = `
<summary>
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
</summary>
`;

exports[`summary renders with accessKey="a,b,c,d" 1`] = `
<summary
  accessKey="a,b,c,d"
/>
`;

exports[`summary renders with autoCapitalize="characters" 1`] = `
<summary
  autoCapitalize="characters"
/>
`;

exports[`summary renders with autoCapitalize="none" 1`] = `
<summary
  autoCapitalize="none"
/>
`;

exports[`summary renders with autoCapitalize="off" 1`] = `
<summary
  autoCapitalize="off"
/>
`;

exports[`summary renders with autoCapitalize="on" 1`] = `
<summary
  autoCapitalize="on"
/>
`;

exports[`summary renders with autoCapitalize="sentences" 1`] = `
<summary
  autoCapitalize="sentences"
/>
`;

exports[`summary renders with autoCapitalize="words" 1`] = `
<summary
  autoCapitalize="words"
/>
`;

exports[`summary renders with children 1`] = `
<summary>
  TEST
</summary>
`;

exports[`summary renders with className="foo bar baz" 1`] = `
<summary
  className="foo bar baz"
/>
`;

exports[`summary renders with contentEditable="false" 1`] = `
<summary
  contentEditable="false"
/>
`;

exports[`summary renders with contentEditable="true" 1`] = `
<summary
  contentEditable="true"
/>
`;

exports[`summary renders with dir="auto" 1`] = `
<summary
  dir="auto"
/>
`;

exports[`summary renders with dir="ltr" 1`] = `
<summary
  dir="ltr"
/>
`;

exports[`summary renders with dir="rtl" 1`] = `
<summary
  dir="rtl"
/>
`;

exports[`summary renders with draggable="false" 1`] = `
<summary
  draggable="false"
/>
`;

exports[`summary renders with draggable="true" 1`] = `
<summary
  draggable="true"
/>
`;

exports[`summary renders with empty attributes 1`] = `<summary />`;

exports[`summary renders with hidden=false 1`] = `
<summary
  hidden={false}
/>
`;

exports[`summary renders with hidden=true 1`] = `
<summary
  hidden={true}
/>
`;

exports[`summary renders with id="TEST" 1`] = `
<summary
  id="TEST"
/>
`;

exports[`summary renders with inputMode="decimal" 1`] = `
<summary
  inputMode="decimal"
/>
`;

exports[`summary renders with inputMode="email" 1`] = `
<summary
  inputMode="email"
/>
`;

exports[`summary renders with inputMode="none" 1`] = `
<summary
  inputMode="none"
/>
`;

exports[`summary renders with inputMode="numeric" 1`] = `
<summary
  inputMode="numeric"
/>
`;

exports[`summary renders with inputMode="search" 1`] = `
<summary
  inputMode="search"
/>
`;

exports[`summary renders with inputMode="tel" 1`] = `
<summary
  inputMode="tel"
/>
`;

exports[`summary renders with inputMode="text" 1`] = `
<summary
  inputMode="text"
/>
`;

exports[`summary renders with inputMode="url" 1`] = `
<summary
  inputMode="url"
/>
`;

exports[`summary renders with is="foo" 1`] = `
<summary
  is="foo"
/>
`;

exports[`summary renders with itemID="https://google.com/" 1`] = `
<summary
  itemID="https://google.com/"
/>
`;

exports[`summary renders with itemProp="https://google.com/" 1`] = `
<summary
  itemProp="https://google.com/"
/>
`;

exports[`summary renders with itemRef="foo bar" 1`] = `
<summary
  itemRef="foo bar"
/>
`;

exports[`summary renders with itemScope=false 1`] = `
<summary
  itemScope={false}
/>
`;

exports[`summary renders with itemScope=true 1`] = `
<summary
  itemScope={true}
/>
`;

exports[`summary renders with itemType="https://google.com/" 1`] = `
<summary
  itemType="https://google.com/"
/>
`;

exports[`summary renders with lang="" 1`] = `
<summary
  lang=""
/>
`;

exports[`summary renders with lang="es-419" 1`] = `
<summary
  lang="es-419"
/>
`;

exports[`summary renders with slot="TEST" 1`] = `
<summary
  slot="TEST"
/>
`;

exports[`summary renders with spellCheck="false" 1`] = `
<summary
  spellCheck="false"
/>
`;

exports[`summary renders with spellCheck="true" 1`] = `
<summary
  spellCheck="true"
/>
`;

exports[`summary renders with style={"color":"#000000"} 1`] = `
<summary
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`summary renders with tabIndex=1 1`] = `
<summary
  tabIndex={1}
/>
`;

exports[`summary renders with title="TEST" 1`] = `
<summary
  title="TEST"
/>
`;

exports[`summary renders with translate="no" 1`] = `
<summary
  translate="no"
/>
`;

exports[`summary renders with translate="yes" 1`] = `
<summary
  translate="yes"
/>
`;

exports[`summary renders without attributes 1`] = `<summary />`;
